src/lib/utils.py

```diff
@@ -418,35 +419,58 @@ def read_json_manifest(manifest_path):
 _yaml.indent(mapping=2, sequence=4, offset=2)
 
 
-def read_yaml_manifest(manifest_path):
+def read_yaml_manifest(manifest_path: Path):
     """Read a YAML manifest from 'manifest_path'."""
-    with open(manifest_path, "r") as f:
+    with manifest_path.open("r") as f:
         return _yaml.load(f)
 
 
-def read_manifest(manifest_path):
+def read_manifest(manifest_path: t.Union[Path, str]):
     """Reads a JSON or YAML manifest from 'manifest_path'."""
-    _, ext = os.path.splitext(manifest_path)
-    if ext in (".yaml", ".yml"):
+    if not isinstance(manifest_path, Path):
+        assert isinstance(manifest_path, str)
+        manifest_path = Path(manifest_path)
+    if manifest_path.suffix in (".yaml", ".yml"):
         return read_yaml_manifest(manifest_path)
     else:
         return read_json_manifest(manifest_path)
 
 
-def dump_manifest(contents, manifest_path):
+def dump_manifest(contents: t.Dict, manifest_path: t.Union[Path, str]):
     """Writes back 'contents' to 'manifest_path'.
 
     For YAML, we make a best-effort attempt to preserve
     formatting; for JSON, we use the canonical 4-space indentation,
     but add a trailing newline if originally present."""
-    _, ext = os.path.splitext(manifest_path)
-    with open(manifest_path, "r") as fp:
-        newline = _check_newline(fp)
-    with open(manifest_path, "w", encoding="utf-8") as fp:
-        if ext in (".yaml", ".yml"):
+    if not isinstance(manifest_path, Path):
```

Comment:
This is probably unnecessary check. You could just call Path(manifest_path) even if given Path to begin with.

Reply:
Indeed, thanks.